### ISSUE TITLE
Cannot add content or upload files to dotCMS

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#11097]
An exception is being thrown when trying to add content to dotCMS.

## Expected Behavior
The system should allow users to create and upload files to dotCMS.

## Current Behavior
An exception is being thrown in the ``dotcms.log`` file:

```
[22/03/17 10:20:34:485 CST]  INFO guava.GuavaCache: ***	 Building Cache : permissioncache, size:50000,Concurrency:32
[22/03/17 10:20:39:373 CST]  WARN cache.LegacyContentTypeCacheImpl: Structure with inode: '' not found in db
[22/03/17 10:20:39:373 CST]  WARN action.EditContentletAction: java.lang.NullPointerException
java.lang.NullPointerException
	at com.dotmarketing.portlets.contentlet.action.EditContentletAction._retrieveWebAsset(EditContentletAction.java:849) ~[dotcms_4.1.0_e9903ec.jar:?]
	at com.dotmarketing.portlets.contentlet.action.EditContentletAction.processAction(EditContentletAction.java:247) [dotcms_4.1.0_e9903ec.jar:?]
	at com.liferay.portal.struts.PortletRequestProcessor.process(PortletRequestProcessor.java:128) [dotcms_4.1.0_e9903ec.jar:?]
	at com.liferay.portlet.StrutsPortlet.processAction(StrutsPortlet.java:116) [dotcms_4.1.0_e9903ec.jar:?]
	at com.liferay.portlet.CachePortlet.processAction(CachePortlet.java:130) [dotcms_4.1.0_e9903ec.jar:?]
	at com.liferay.portal.action.LayoutAction._processPortletRequest(LayoutAction.java:232) [dotcms_4.1.0_e9903ec.jar:?]
	at com.liferay.portal.action.LayoutAction._processActionRequest(LayoutAction.java:249) [dotcms_4.1.0_e9903ec.jar:?]
	at com.liferay.portal.action.LayoutAction.execute(LayoutAction.java:93) [dotcms_4.1.0_e9903ec.jar:?]
	at com.dotcms.repackage.org.apache.struts.action.RequestProcessor.processActionPerform(RequestProcessor.java:421) [dot.struts-1.2.4_2.jar:1.2.4]
	at com.dotcms.repackage.org.apache.struts.action.RequestProcessor.process(RequestProcessor.java:226) [dot.struts-1.2.4_2.jar:1.2.4]
	at com.liferay.portal.struts.PortalRequestProcessor.process(PortalRequestProcessor.java:128) [dotcms_4.1.0_e9903ec.jar:?]
	at com.dotcms.repackage.org.apache.struts.action.ActionServlet.process(ActionServlet.java:1164) [dot.struts-1.2.4_2.jar:1.2.4]
	at com.dotcms.repackage.org.apache.struts.action.ActionServlet.doGet(ActionServlet.java:397) [dot.struts-1.2.4_2.jar:1.2.4]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:618) [servlet-api.jar:?]
	at com.liferay.portal.servlet.MainServlet.callParentService(MainServlet.java:281) [dotcms_4.1.0_e9903ec.jar:?]
	at com.liferay.portal.servlet.MainServlet.service(MainServlet.java:510) [dotcms_4.1.0_e9903ec.jar:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:725) [servlet-api.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:291) [catalina.jar:8.0.18]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) [tomcat-websocket.jar:8.0.18]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
	at com.dotcms.repackage.com.liferay.filters.secure.SecureFilter.doFilter(SecureFilter.java:135) [dot.secure-filter-ukv_2.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
	at com.dotmarketing.filters.CMSFilter.doFilter(CMSFilter.java:265) [dotcms_4.1.0_e9903ec.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:73) [dotcms_4.1.0_e9903ec.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
	at com.dotmarketing.cms.urlmap.filters.URLMapFilter.doFilter(URLMapFilter.java:308) [dotcms_4.1.0_e9903ec.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
	at com.dotcms.repackage.org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:404) [dot.urlrewritefilter-4.0.3_2.jar:4.0.3]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
	at com.dotmarketing.filters.TimeMachineFilter.doFilter(TimeMachineFilter.java:134) [dotcms_4.1.0_e9903ec.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
	at com.dotmarketing.filters.ThreadNameFilter.doFilter(ThreadNameFilter.java:90) [dotcms_4.1.0_e9903ec.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
	at com.dotmarketing.filters.CookiesFilter.doFilter(CookiesFilter.java:33) [dotcms_4.1.0_e9903ec.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
	at com.dotmarketing.filters.CharsetEncodingFilter.doFilter(CharsetEncodingFilter.java:110) [dotcms_4.1.0_e9903ec.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:219) [catalina.jar:8.0.18]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106) [catalina.jar:8.0.18]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:501) [catalina.jar:8.0.18]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:142) [catalina.jar:8.0.18]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79) [catalina.jar:8.0.18]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:610) [catalina.jar:8.0.18]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88) [catalina.jar:8.0.18]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:516) [catalina.jar:8.0.18]
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1086) [tomcat-coyote.jar:8.0.18]
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:659) [tomcat-coyote.jar:8.0.18]
	at org.apache.coyote.http11.Http11NioProtocol$Http11ConnectionHandler.process(Http11NioProtocol.java:223) [tomcat-coyote.jar:8.0.18]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1558) [tomcat-coyote.jar:8.0.18]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1515) [tomcat-coyote.jar:8.0.18]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_65]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_65]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.0.18]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_65]
```

## Possible Solution
Use the new Content Type API to retrieve Content Type objects.

## Known Workarounds
None.

## Steps to Reproduce (for bugs)
1. Log into dotCMS 4.1
2. Go to the Content > Search page.
3. Try to create a content with any Content Type.
4. The message: "An unexpected system error occurred." is displayed.

## Your Environment
* dotCMS version used: 4.1
* Browser Name and version: Any
* Operating System and version: Any
* Application Server and version: Tomcat 8
* Java Brand and version: Java 8
* Database and version: Any
* Application module (if apply): Site Browser and Content Search pages.

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/11144

**Comment by bryanboza** [comment]
Fixed, tested on master-4.1 // Postgres
